CodeEdit/Features/SourceControl/Client/GitClient+Pull.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  GitClient+Pull.swift
+//  CodeEdit
+//
+//  Created by Austin Condiff on 11/18/23.
+//
+
+import Foundation
+
+extension GitClient {
+    /// Pull changes from remote
+    func pullFromRemote(remote: String? = nil, branch: String? = nil, rebase: Bool? = nil) async throws {
+        var command = "pull"
+
+        if let remote = remote, let branch = branch {
+            command += " \(remote) \(branch)"
+        }
+
+        if rebase == true {
```

Comment:
Why are you checking if it's explicitly true?

Reply:
Because it might be either `nil` or `false`.

Reply:
It never needs to be nil only false or true, i would opt for always returning false instead of true

Reply:
Done